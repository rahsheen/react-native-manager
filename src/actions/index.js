import firebase from 'firebase'
import { google } from 'react-native-simple-auth'
import {
    EMAIL_CHANGED,
    PASSWORD_CHANGED,
    LOGIN_USER_SUCCESS,
    LOGIN_USER_CREDENTIALS,
    LOGIN_USER_FAIL,
    LOGIN_USER
} from './types'

export const emailChanged = (text) => {
    return {
        type: EMAIL_CHANGED,
        payload: text
    }
}

export const passwordChanged = (text) => {
    return {
        type: PASSWORD_CHANGED,
        payload: text
    }
}

export const loginUser = ({ email, password }) => {
    return (dispatch) => {
        dispatch({ type: LOGIN_USER })

        firebase.auth().signInWithEmailAndPassword(email, password)
            .then(user => loginUserSuccess(dispatch, user))
            .catch(() => {
                firebase.auth().createUserWithEmailAndPassword(email, password)
                    .then(user => loginUserSuccess(dispatch, user))
                    .catch(() => loginUserFail(dispatch))
            })
    }
}

const loginUserSuccess = (dispatch, user) => {
    dispatch({
        type: LOGIN_USER_SUCCESS,
        payload: user
    })
}

const loginUserFail = (dispatch) => {
    dispatch({ type: LOGIN_USER_FAIL })
}

export const googleLogin = () => {
    return dispatch => {
        dispatch({ type: LOGIN_USER })

        google({
            appId: '484896703731-s31cjcqcd5put1eev7g6dhn2jsqcpvda.apps.googleusercontent.com',
            callback: 'biz.sheenonline:/oauth2redirect',
        }).then(info => {
            // info.user - user details from the provider
            // info.pcredentials - tokens from the provider
            loginUserSuccess(dispatch, info.user)

            dispatch({
                type: LOGIN_USER_CREDENTIALS,
                payload: info.credentials
            })

        }).catch(error => {
            // error.code
            // error.description
            console.log(error.description)
        });

    }
}