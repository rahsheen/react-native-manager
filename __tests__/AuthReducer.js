import authReducer from '../src/reducers/AuthReducer'
import { EMAIL_CHANGED } from '../src/actions/types'

it('returns a new state', () => {
    const INITIAL_STATE = { email: 'boo' } 
    const theAction = { type: EMAIL_CHANGED, email: 'blah@foo.com' }

    expect(authReducer(INITIAL_STATE, theAction)).toMatchSnapshot()
})