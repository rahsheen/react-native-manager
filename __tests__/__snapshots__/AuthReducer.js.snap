// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns a new state 1`] = `
Object {
  "email": undefined,
}
`;
